subsys/net/lib/azure_fota/azure_fota.c

```diff
@@ -23,52 +23,78 @@
 
 LOG_MODULE_REGISTER(azure_fota, CONFIG_AZURE_FOTA_LOG_LEVEL);
 
-#define STATUS_CURRENT_STR "\"currentFwVersion\":\"%s\""
-#define STATUS_PENDING_STR "\"pendingFwVersion\":\"%s\""
-#define STATUS_UPDATE_STR "\"fwUpdateStatus\":\"%s\""
-#define STATUS_UPDATE_SUB_STR "\"fwUpdateSubstatus\":\"%s\""
+#define STATUS_CURRENT_STR	"\"currentFwVersion\":\"%s\""
+#define STATUS_PENDING_STR	"\"pendingFwVersion\":\"%s\""
+#define STATUS_UPDATE_STR	"\"fwUpdateStatus\":\"%s\""
+#define STATUS_UPDATE_SUB_STR	"\"fwUpdateSubstatus\":\"%s\""
+#define STATUS_JOB_ID		"\"jobId\":\"%s\""
 
 #define STATUS_REPORT					\
 	"{"						\
 		"\"firmware\":{"			\
 			STATUS_UPDATE_STR ","		\
-			STATUS_CURRENT_STR ","		\
-			STATUS_PENDING_STR		\
+			STATUS_CURRENT_STR		\
 		"}"					\
 	"}"
 
-/* Enum to keep the fota status */
-enum fota_status {
-	STATUS_CURRENT = 0,
-	STATUS_DOWNLOADING,
-	STATUS_APPLYING,
-	STATUS_REBOOTING,
-	STATUS_ERROR,
-	STATUS_ROLLEDBACK,
-
-	STATUS_COUNT,
-};
+#define STATUS_REPORT_UPDATING					\
+	"{"						\
+		"\"firmware\":{"			\
+			STATUS_UPDATE_STR ","		\
+			STATUS_CURRENT_STR ","		\
+			STATUS_PENDING_STR ","		\
+			STATUS_JOB_ID			\
+		"}"					\
+	"}"
 
-struct fota_object {
+/* Enum to keep track of the internal state. */
+static enum internal_state {
+	/* Library is uninitialized, no messages can be processed. */
+	STATE_UNINIT,
+	/* Initialized and waiting for incoming messages to process. */
+	STATE_INIT,
+	/* Downloading FOTA image. */
+	STATE_DOWNLOADING,
+} internal_state = STATE_UNINIT;
+
+/* Enum to keep the reported fota status */
+static enum fota_status {
+	REP_STATUS_CURRENT,
+	REP_STATUS_DOWNLOADING,
+	REP_STATUS_APPLYING,
+	REP_STATUS_REBOOTING,
+	REP_STATUS_ERROR,
+	REP_STATUS_ROLLEDBACK,
+
+	REP_STATUS_COUNT,
+} current_status = REP_STATUS_CURRENT;
+
+static struct fota_object {
 	char host[CONFIG_AZURE_FOTA_HOSTNAME_MAX_LEN];
 	char path[CONFIG_AZURE_FOTA_FILE_PATH_MAX_LEN];
 	char version[CONFIG_AZURE_FOTA_VERSION_MAX_LEN];
+	char job_id[CONFIG_AZURE_FOTA_JOB_ID_MAX_LEN];
 	size_t fragment_size;
-};
+} fota_object;
 
 /* Static variables */
```

Comment:
This line can be moved to include ```internal_state```, ```fota_status```, and ```fota_object```. I guess they are also static variables since they are declared in this case.